pkg/controllers/addons_controller.go

```diff
@@ -39,224 +36,120 @@ import (
 // AddonsLayerReconciler reconciles a AddonsLayer object.
 type AddonsLayerReconciler struct {
 	client.Client
-	Log     logr.Logger
-	Scheme  *runtime.Scheme
-	Applier apply.LayerApplier
+	Config   *rest.Config
+	k8client kubernetes.Interface
```

Comment:
@paulcarlton-ww  @sbernheim @richardcase I am think passing on a *rest.Config and kubernetes.Interface might not be a good idea. It might encourage usage of kubernetes.Interface (bcs it is available) rather than using controller-runtime Client. If the sole purpose of this is for fetching the server version (unless I am missing something), what do you guys thinking about embedding DiscoveryInterface adding just the discovery behaviour to the reconciler, 
type AddonsLayerReconciler struct {
	client.Client
	discovery.DiscoveryInterface

This way, while constructing the NewAddonsLayerReconciler, it can directly take this interface as input and we can stay away from depending on *rest.Config. Thoughts?

Reply:
I think we should leave it as is for now and look at this again later, as Kraan evolves we may find other things we need the full K8s client for. Have raised issue for this https://github.com/fidelity/kraan/issues/18